snappi_ixnetwork/trafficitem.py

```diff
@@ -203,34 +598,59 @@ def config(self):
         - CREATE TrafficItem for any config.flows[*].name that does not exist
         - UPDATE TrafficItem for any config.flows[*].name that exists
         """
-        ixn_traffic_item = self._api._traffic_item
-        self.flows_has_latency = []
-        self.flows_has_timestamp = []
-        self.flows_has_loss = []
-        self.latency_mode = None
-        self._api._remove(ixn_traffic_item, self._api.snappi_config.flows)
-        if len(self._api.snappi_config.flows) > 0:
-            for flow in self._api.snappi_config.flows:
-                self._endpoint_validation(flow)
-                args = {
-                    "Name": flow.name,
-                    "TrafficItemType": "l2L3",
-                    "TrafficType": self._get_traffic_type(flow),
-                    "SrcDestMesh": self._get_mesh_type(flow),
-                }
-                ixn_traffic_item.find(
-                    Name="^%s$" % self._api.special_char(flow.name)
+        if len(self._api.snappi_config.flows) == 0:
+            self.remove_ixn_traffic()
+            return
+        with Timer(self._api, "Flows configuration"):
+            self.remove_ixn_traffic()
+            ixn_traffic_item = self.get_ixn_config(self._api.snappi_config)[0]
+            self.flows_has_latency = []
+            self.flows_has_timestamp = []
+            self.flows_has_loss = []
+            self.latency_mode = None
+            if ixn_traffic_item.get("trafficItem") is None:
+                # TODO raise Exception
+                return
+            ixn_traffic_item = ixn_traffic_item.get("trafficItem")
+            tr_json = {"traffic": {"xpath": "/traffic", "trafficItem": []}}
+            for i, flow in enumerate(self._api.snappi_config.flows):
+                tr_item = {"xpath": ixn_traffic_item[i]["xpath"]}
+                if ixn_traffic_item[i].get("configElement") is None:
+                    raise Exception(
+                        "Endpoints are not properly configured in IxNetwork"
+                    )
+                ce_xpaths = [
+                    {"xpath": ce["xpath"]}
+                    for ce in ixn_traffic_item[i]["configElement"]
+                ]
+                tr_item["configElement"] = ce_xpaths
+                self._configure_size(
+                    tr_item["configElement"], flow.get("size")
                 )
-                if len(ixn_traffic_item) == 0:
-                    ixn_traffic_item.add(**args)
-                else:
-                    self._update(ixn_traffic_item, **args)
-                self._configure_endpoint(
-                    ixn_traffic_item.EndpointSet, flow.tx_rx
+                self._configure_rate(
```

Comment:
get with_default=True for rate, size and duration. Otherwise it will configure IxNetwork default. And I think we are decide to configure with snappi default value

Reply:
I remember adding the defaults. let me check again